### PR TITLE
[NTOS:KE] Improve KeConnectInterrupt

### PR DESCRIPTION
**Pull request by tkreuzer**
"Good" enough to not hit a break point.

### CODE REVIEW

File: ntoskrnl/ke/amd64/interrupt.c [hunk 1]
```diff
@@ -85,14 +85,24 @@ KeConnectInterrupt(IN PKINTERRUPT Interrupt)
...
     ASSERT(Interrupt->Irql <= HIGH_LEVEL);
+    ASSERT(Interrupt->SynchronizeIrql >= Interrupt->Irql);
+    ASSERT(Interrupt->Irql == (Interrupt->Vector >> 4));
+
+    /* Make sure the vector is within the allowed range */
+    if (Interrupt->Vector < PRIMARY_VECTOR_BASE)
+    {
+        DPRINT1("FIXME: KeConnectInterrupt() with invalid vector 0x%lx!\n", Interrupt->Vector);
+        //__debugbreak();
+        //return FALSE;
```
Comment (zefklop) [inline ntoskrnl/ke/amd64/interrupt.c:96]: did you actually hit this case ?

**Comment by zefklop** [comment]
Ok for me. Can you delete the commented-out debug breaks ? Keeping them just makes things confusing.

**Review by zefklop — APPROVED** [review]

File: ntoskrnl/ke/amd64/interrupt.c (hunk 1 above)
Comment (tkreuzer) [reply ntoskrnl/ke/amd64/interrupt.c:96]: Yes, it's broken 😄 I'll fix it, once everything is merged. Or maybe The_DarkFire fixes it first.

Comment (tkreuzer) [reply ntoskrnl/ke/amd64/interrupt.c]: It's no longer broken!

**Pull request merged**